### DISCUSSION TITLE
Integrate with AutoCAD

### DISCUSSION THREAD

**Discussion by jtoverka** [leppie/IronScheme#124] [Discussions] [answered]
I am trying to integrate this library IronScheme.dll as a powerful scripting engine for AutoCAD. However, I am running into an error:
ArgumentException - Unknown language identifier

Here is the code that won't work:

```
using IronScheme.Scripting;
using System.IO;

/// <summary>
/// Executes the file as a Scheme script.
/// </summary>
/// <param name="file">The file with the Scheme script.</param>
private static void ExecuteSchemeScript(string file)
{
	if (File.Exists(file))
	{
		Script.ExecuteFile(file);
	}
}
```

I just want to be able to execute script files to get started.

Looking through the code here, it runs into the issue below:
The _languageIds do not have anything inside the collection. And it will always throw an exception when I try to execute a file. What am I doing wrong in my setup?
![image](https://user-images.githubusercontent.com/22569118/167459763-ce3b6130-dc24-4e4b-9610-8b7796f06f4e.png)

I have compiled the latest release of IronScheme and used IronScheme.Scripting.dll and it's dependency Oyster.IntX.dll. I used those as references and it resulted in the same exception.

**Comment by jtoverka** [comment, marked as answer]
I found the issue, I needed to string.eval and have the IronScheme.dll netloaded as well as my app.

**Comment by leppie** [comment]
You probably want to steer away from any DLR concepts. 

But as you see, just do `Eval` and you good :D
